api/apps/v1alpha1/nemo_guardrails_types.go

```
@@ -81,11 +81,19 @@ type NemoGuardrailSpec struct {
 	RuntimeClass string `json:"runtimeClass,omitempty"`
 }
 
+// GuardrailConfig defines the source where the service config is made available.
+//
+// +kubebuilder:validation:XValidation:rule="!(has(self.configMap) && has(self.pvc))", message="Cannot set both ConfigMap and PVC in ConfigStore"
 type GuardrailConfig struct {
-	ConfigMap string                 `json:"configMap,omitempty"`
+	ConfigMap *ConfigMap             `json:"configMap,omitempty"`
```

ConfigMap was used for testing purposes and can be removed. Guardrail config can be updated dynamically through the API and those will be lost if we are using a CM. @slu2011 thoughts?

---

I dont mind either way. Although the CM makes for a quick way to test the CR since it comes pre-populated with a config.

---

Ok, since CM is always mounted as `read-only` any POST call to update the config should return an error and we can document that clearly.